File: src/lib/ipc/audio.ts
import { invoke } from '@tauri-apps/api/core';
import type { AudioMode, VolumeKind } from '$lib/types';

export const audioSetMode = (mode: AudioMode) =>
  invoke<void>('audio_set_mode', { mode });

export const audioSetVolume = (kind: VolumeKind, value: number) =>
  invoke<void>('audio_set_volume', { kind, value });

File: src-tauri/src/commands/audio.rs
//! Audio commands
//!
//! TODO(backend): `core::audio_service::AudioService` 経由で rodio を操作

use crate::error::AppResult;
use crate::models::AudioMode;

#[tauri::command]
pub fn audio_set_mode(mode: AudioMode) -> AppResult<()> {
    log::info!("[stub] audio_set_mode called: {:?}", mode);
    Ok(())
}

#[tauri::command]
pub fn audio_set_volume(kind: String, value: f32) -> AppResult<()> {
    log::info!("[stub] audio_set_volume called: {}={}", kind, value);
    // TODO(backend): kind を "master" | "water" | "kakon" で分岐
    Ok(())
}
